internal/net/tcp/dialer.go

```diff
@@ -50,6 +53,27 @@ type dialer struct {
 	dialer                func(ctx context.Context, network, addr string) (net.Conn, error)
 }
 
+type dialerCache struct {
+	ips []string
+	cnt uint32
+}
+
+// GetIP returns the next cached IP address in round robin order.
+// It starts getting the index 1 cache instead of index 0 cache.
+func (d *dialerCache) GetIP() string {
```

Comment:
could you please change function naming to IP()

Reply:
fixed :)